### PR TITLE
Enable kernel debugs for rt job

### PR DESCRIPTION
**Pull request by mrunalp**
systemd team needs kernel debug enabled to look into the
issues we are hitting in the rt jobs

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by mrunalp** [comment]
@cgwalters @sjenning @deads2k ptal

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 1]
```diff
@@ -852,6 +852,30 @@ objects:
...
+          cat > ${manifests}/worker-kernel-debug-machine-config.yml << EOF
+        apiVersion: machineconfiguration.openshift.io/v1
+        kind: MachineConfig
+        metadata:
+          labels:
+            machineconfiguration.openshift.io/role: worker
+          name: kernel-debug
+        spec:
+          kernelArguments:
+          - 'debug'
```
Comment (cgwalters) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml:875]: This looks likely to do something, I see
`debug		[KNL] Enable kernel debugging (events log level).`
but...are you sure the person isn't asking for `kernel-debug` which is an entirely separate kernel build that we're not including right now?  See https://fedoraproject.org/wiki/KernelDebugStrategy

Comment (cgwalters) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml:875]: Oh I see
```
       debug
           Turn on debugging output. This is equivalent to systemd.log_level=debug. Note that this option is also read by the kernel itself and enables
           kernel debug output. Passing this option hence turns on the debug output from both the system manager and the kernel.
```

**Review by cgwalters — APPROVED** [review]
Seems OK to me although a few thoughts (not blocking):

- Conceptually enabling debug for systemd/kernel is orthogonal to rt, so this could be a separate variant `kernel-debug-log` that could be used with non-rt too
- Do we really need a whole cluster with this or wouldn't it have been sufficient to just `rpm-ostree kargs --append=debug` on a single worker node and reboot?

**Comment by wking** [comment]
/lgtm

I don't understand the trade-offs for one MachineConfig entry vs. two MachineConfig entries, but if @mrunalp and @cgwalters are fine with two, that's good enough for me.  Would be nice to have this job ported to multi-step soon so that the machine-config folks could be their own maintainers for this content.

**Pull request merged**